shell/src/main/java/alluxio/cli/UnderFileSystemContractTest.java

```diff
@@ -111,42 +106,47 @@ private void runS3Operations() throws Exception {
     mUfs = UnderFileSystem.Factory.create(mUfsPath, getUfsConf());
 
     String testDir = createTestDirectory();
-    loadAndRunTests(new S3ASpecificOperations(testDir, mUfs, mConf), testDir);
+    return loadAndRunTests(new S3ASpecificOperations(testDir, mUfs, mConf), testDir);
   }
 
   /**
    * Loads and runs the tests in the target operations class.
    *
    * @param operations the class that contains the tests to run
    * @param testDir the test directory to run tests against
+   * @return the number of failed tests
    */
-  private void loadAndRunTests(Object operations, String testDir) throws Exception {
+  private int loadAndRunTests(Object operations, String testDir) throws Exception {
+    int failedTestCnt = 0;
     try {
       Class classToRun = operations.getClass();
-      Field[] fields = classToRun.getDeclaredFields();
-      for (Field field : fields) {
-        field.setAccessible(true);
-      }
       Method[] tests = classToRun.getDeclaredMethods();
       for (Method test : tests) {
         String testName = test.getName();
         if (testName.endsWith("Test")) {
           System.out.printf("Running test: %s...", testName);
+          boolean passed = false;
           try {
             test.invoke(operations);
-          } catch (InvocationTargetException e) {
+            passed = true;
+            cleanupUfs(testDir);
+          } catch (Exception e) {
```

Comment:
we should avoid catch-all here. Is there a reason to use `Exception` over InvocationTargetException?

Reply:
I think it makes sense to let all the tests complete, instead of letting one ufs operation err out and aborts the `runUfsTests`.